lib/arpv4/arpv4.ml

```diff
@@ -175,7 +153,7 @@ module Make (Ethif : V1_LWT.ETHIF) (Clock : V1.CLOCK) (Time : V1_LWT.TIME) = str
     (* Source protocol address, pick one of our IP addresses *)
     let spa = match t.bound_ips with
       | hd::_ -> hd | [] -> Ipaddr.V4.any in
-    output t { op=`Request; tha; sha; tpa; spa }
+    output t Arpv4_parse.({ op=Arpv4_wire.Request; tha; sha; tpa; spa })
```

Comment:
It looks a bit awkward using the parser module's type to generate output. Perhaps parsing and printing should be in a single `Arpv4_msg` module? In an ideal world, we might even generate the parser and the printer from a single definition.